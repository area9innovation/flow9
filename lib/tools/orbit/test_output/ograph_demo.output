Flow compiler (3rd generation)

Processing 'orbit' on http server

Created ograph: demo
Added node: 1
Variable not found: x
Variable not found: y
Added node: 2
TIMEOUT after 10 seconds
