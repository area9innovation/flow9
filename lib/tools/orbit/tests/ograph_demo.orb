// Demo of OGraph functionality

// Create a new ograph
let graph = makeOGraph("demo");
println("Created ograph: " + graph);

// Create some expressions to add to the graph

// Add simple expression: 2 + 3
let expr1 = 2 + 3;
let node1 = addNodeToOGraph(graph, "math", expr1);
println("Added node: " + i2s(node1));

// Add another expression: x * y
let expr2 = x * y;
let node2 = addNodeToOGraph(graph, "math", expr2);
println("Added node: " + i2s(node2));

// Add a domain annotation
let addDomain1 = addDomainToNode(graph, node1, "Algebra");
println("Added domain: " + b2s(addDomain1));

// Add another domain annotation
let addDomain2 = addDomainToNode(graph, node2, "Algebra");
println("Added domain: " + b2s(addDomain2));

// Add another domain annotation to the same node
let addDomain3 = addDomainToNode(graph, node2, "Sâ‚‚");  // Symmetric group of order 2
println("Added domain: " + b2s(addDomain3));

// Print the graph
println("\nGraph before merging:");
printOGraph(graph);

// Create and add a new expression to merge with an existing one
let expr3 = 2 + 3;  // Same as expr1
let node3 = addNodeToOGraph(graph, "math", expr3);
println("Added redundant node: " + i2s(node3));

// Merge the nodes
let merged = mergeOGraphNodes(graph, node1, node3);
println("Merged nodes: " + b2s(merged));

// Print the graph after merging
println("\nGraph after merging:");
printOGraph(graph);

// Extract a node back to an Orbit expression
let extracted = extractOGraphNode(graph, node1);
println("\nExtracted node " + i2s(node1) + " as expression: " + i2s(eval(extracted)));

// Create a more complex expression for pattern matching demo
let pattern = a + b : ast;
let node4 = addNodeToOGraph(graph, "pattern", pattern);
println("\nAdded pattern node: " + i2s(node4));

// Print the final graph
println("\nFinal graph:");
printOGraph(graph)