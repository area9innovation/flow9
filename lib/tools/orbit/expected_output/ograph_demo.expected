Flow compiler (3rd generation)

Processing 'orbit' on http server

Created ograph: demo
Added node: 1
Variable not found: x
Variable not found: y
Added node: 2
Variable not found: b2s
Cannot call non-function value: 0
Expected int, double, string or array for +, not true
Added domain: 
Variable not found: b2s
Cannot call non-function value: 0
Expected int, double, string or array for +, not true
Added domain: 
Variable not found: b2s
Cannot call non-function value: 0
Expected int, double, string or array for +, not true
Added domain: 

Graph before merging:
Added redundant node: 6
Variable not found: b2s
Cannot call non-function value: 0
Expected int, double, string or array for +, not true
Merged nodes: 

Graph after merging:
ERROR: Expected int, not "expr1"

Extracted node 1 as expression: 0
Variable not found: a
TODO: Interpret OrTypeAnnotation: b : ast

Added pattern node: 7

Final graph:
Result: "Node 1 [math]: OrIdentifier\nNode 2 [math]: OrIdentifier\nNode 3 [meta]: domain\nNode 4 [meta]: domain\nNode 5 [meta]: domain\nNode 1 [math]: OrIdentifier\nNode 7 [pattern]: OrIdentifier\n"
VERIFY - THIS OUTPUT NEEDS HUMAN VERIFICATION
